### PR TITLE
[FLOC-3033] Stop things before removing them

### PR DESCRIPTION
**Pull request by exarkun**
Fixes https://clusterhq.atlassian.net/browse/FLOC-3033

### CODE REVIEW

File: flocker/provision/_install.py [hunk 1]
```diff
@@ -980,14 +980,31 @@ def _uninstall_flocker_centos7():
...
+            u"{{ "
+            u"systemctl is-enabled {unit} && "
+            u"systemctl stop {unit} && "
+            u"systemctl disable {unit} "
+            u"; }} || /bin/true".format(unit=unit).encode("ascii")
+        )
+
+    return sequence(
+        list(
+            # XXX There should be uninstall hooks for stopping services.
```
Comment (adamtheturtle) [inline flocker/provision/_install.py]: I'll have a look at this tomorrow but it looks like this (or the XXX follow up) could borrow from http://doc-dev.clusterhq.com/gettinginvolved/contributing.html?highlight=test#testing-code-on-nodes

Comment (wallrj) [inline flocker/provision/_install.py]: Here's some stuff about post install RPM hooks: http://www.ibm.com/developerworks/library/l-rpm2/

**Comment by wallrj** [comment]
Forced rebuild of failed builds.

File: flocker/provision/_install.py (hunk 1 above)
Comment (Azulinho) [inline flocker/provision/_install.py]: we use fpm don't we ?
it supports pre/post hooks

```
--after-install FILE          A script to be run after package installation
    --before-install FILE         A script to be run before package installation
    --after-remove FILE           A script to be run after package removal
    --before-remove FILE          A script to be run before package removal
    --after-upgrade FILE          A script to be run after package upgrade. If not specified,
                                  --before-install, --after-install, --before-remove, and 
                                  --after-remove wil behave in a backwards-compatible manner
                                  (they will not be upgrade-case aware).
                                  Currently only supports deb and rpm packages.
    --before-upgrade FILE         A script to be run before package upgrade. If not specified,
                                  --before-install, --after-install, --before-remove, and 
                                  --after-remove wil behave in a backwards-compatible manner
                                  (they will not be upgrade-case aware).
                                  Currently only supports deb and rpm packages.
```

**Comment by exarkun** [comment]
Builds are passing now.

**Comment by adamtheturtle** [comment]
Thanks, merge if you have run this and it is removing the dangling units.

**Pull request merged**